subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java

```diff
@@ -135,7 +137,12 @@ public FileWatchingStatistics getAndResetStatistics() {
     public void close() throws IOException {
         stopping = true;
         try {
-            watcher.close();
+            watcher.shutdown();
+            if (!watcher.awaitTermination(5, TimeUnit.SECONDS)) {
+                throw new RuntimeException("Watcher did not terminate withing 5 seconds");
+            }
+        } catch (InterruptedException e) {
+            throw new InterruptedIOException("Awaiting termination of watcher was interrupted");
```

Comment:
Do we need to interrupt the thread here?

Reply:
I added the calls to `Thread.currentThread().interrupt()` in https://github.com/gradle/gradle/pull/13003.